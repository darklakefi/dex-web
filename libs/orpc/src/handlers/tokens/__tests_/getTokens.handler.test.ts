import { getTokensHandler } from "../getTokens.handler";

describe("getTokensHandler", () => {
  it.skip("should return a list of assets", async () => {
    const result = await getTokensHandler({
      limit: 10,
      offset: 0,
    });

    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
  });
});
