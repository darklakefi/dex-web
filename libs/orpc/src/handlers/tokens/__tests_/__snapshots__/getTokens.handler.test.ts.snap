// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getTokensHandler > should return a list of assets 1`] = `
{
  "hasMore": true,
  "tokens": [
    {
      "address": "7gxzDSLbXqapoJ1e4WubzWUfFDeZZPENMAfCQeKfYyjT",
      "decimals": 6,
      "imageUrl": "",
      "name": "$F$",
      "symbol": "$F$",
      "value": "7gxzDSLbXqapoJ1e4WubzWUfFDeZZPENMAfCQeKfYyjT",
    },
    {
      "address": "9gXQd53kyGXB1juo7eKpfSTrvCW26u9LfUsPC9HH4nGQ",
      "decimals": 9,
      "imageUrl": "",
      "name": "JOHN",
      "symbol": "JOHN",
      "value": "9gXQd53kyGXB1juo7eKpfSTrvCW26u9LfUsPC9HH4nGQ",
    },
    {
      "address": "21BTCo9hWHjGYYUQQLqjLgDBxjcn8vDt4Zic7TB3UbNE",
      "decimals": 8,
      "imageUrl": "https://cdn.onyx.21.co/images/tokens/21BTC.png",
      "name": "21.co Wrapped Bitcoin",
      "symbol": "21BTC",
      "value": "21BTCo9hWHjGYYUQQLqjLgDBxjcn8vDt4Zic7TB3UbNE",
    },
    {
      "address": "21kbezaCxz9dG2NtXp7pW1hAmtKe6ykiJkeT1EiGXDyY",
      "decimals": 9,
      "imageUrl": "https://i.imgur.com/KPPBoDB.png",
      "name": "Exiled Dust Protocol",
      "symbol": "EDUST",
      "value": "21kbezaCxz9dG2NtXp7pW1hAmtKe6ykiJkeT1EiGXDyY",
    },
    {
      "address": "22513u2QwiY6xaJn7nVFWGKy3aBdw6WfZsRPW2RRtCKj",
      "decimals": 6,
      "imageUrl": "https://cf-ipfs.com/ipfs/Qmdh9Jz5SEPKeRSmTu7Ey3BnFEqAPtGZ9nM6qKceJhk1Jv",
      "name": "Hunter Boden",
      "symbol": "HuntBoden",
      "value": "22513u2QwiY6xaJn7nVFWGKy3aBdw6WfZsRPW2RRtCKj",
    },
    {
      "address": "24a1k6wSDQUSwoRXJyvkr25kF5do8f6Ddff6A2fTc1pM",
      "decimals": 9,
      "imageUrl": "https://arweave.net/7Qzb-BKL734z8w77fPNJ8lu3f6TtA87nbiw2A-qmCK0",
      "name": "BEAR",
      "symbol": "$BEAR",
      "value": "24a1k6wSDQUSwoRXJyvkr25kF5do8f6Ddff6A2fTc1pM",
    },
    {
      "address": "25hAyBQfoDhfWx9ay6rarbgvWGwDdNqcHsXS3jQ3mTDJ",
      "decimals": 5,
      "imageUrl": "https://img.fotofolio.xyz/?url=https%3A%2F%2Fi.ibb.co%2FJnz56gp%2FManeki-PFP.png",
      "name": "MANEKI",
      "symbol": "MANEKI",
      "value": "25hAyBQfoDhfWx9ay6rarbgvWGwDdNqcHsXS3jQ3mTDJ",
    },
    {
      "address": "25p2BoNp6qrJH5As6ek6H7Ei495oSkyZd3tGb97sqFmH",
      "decimals": 6,
      "imageUrl": "https://img.fotofolio.xyz/?url=https%3A%2F%2Fcf-ipfs.com%2Fipfs%2FQmW1VaQ31oKGUeRnXiohvrZXzu1ETBWK3rM4haVYnfShXo",
      "name": "SolBastard",
      "symbol": "soba",
      "value": "25p2BoNp6qrJH5As6ek6H7Ei495oSkyZd3tGb97sqFmH",
    },
    {
      "address": "25yPcyqSpSv8T2JWyoKKRiU622Sr7LoMk9S3FmuBFt8S",
      "decimals": 2,
      "imageUrl": "https://img-cdn.magiceden.dev/rs:fill:400:400:0:0/plain/https://creator-hub-prod.s3.us-east-2.amazonaws.com/gmers_pfp_1662860639809.png",
      "name": "GMers",
      "symbol": "GMERS",
      "value": "25yPcyqSpSv8T2JWyoKKRiU622Sr7LoMk9S3FmuBFt8S",
    },
    {
      "address": "26KMQVgDUoB6rEfnJ51yAABWWJND8uMtpnQgsHQ64Udr",
      "decimals": 6,
      "imageUrl": "https://bafkreidnhl3agzjkay3ljojmvwid72amoumqftgaw7lks2keg27ph3efsu.ipfs.nftstorage.link",
      "name": "SAD HAMSTER",
      "symbol": "HAMMY",
      "value": "26KMQVgDUoB6rEfnJ51yAABWWJND8uMtpnQgsHQ64Udr",
    },
  ],
  "total": 1497,
}
`;
